modules/cluster-ingress/main.tf

```diff
@@ -0,0 +1,40 @@
+resource "kubectl_manifest" "cluster-ingress" {
+  yaml_body = <<YAML
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: ${var.argo_deployment_name}
+  namespace: argocd
+spec:
+  project: default
+  %{if var.auto_deploy}
+  syncPolicy:
+    automated:
+      prune: ${var.auto_prune}
+  %{endif}
+  sources:
+  - repoURL: 'https://github.com/Sage-Bionetworks-Workflows/eks-stack.git'
+    targetRevision: ${var.git_revision}
+    path: modules/cluster-ingress/resources
+    kustomize:
+      patches:
+      - target:
+          kind: Gateway
+        patch: |-
+          - op: replace
+            path: /metadata/annotations/cert-manager.io~1cluster-issuer
+            value: ${var.cluster_issuer_name}
+          - op: replace
+            path: /spec/listeners/0/hostname
+            value: ${var.ssl_hostname}

```

Comment:
This ssl_hostname is a little rough in it's current form. In order to know what the hostname is going to be this module has to be deployed, and we need to get the DNS name of the AWS load balancer that is created, then apply the change again with the correct value.

Ideally, if we get a subdomain to use that is already created we will have this static value ahead of time to pass in here.

Reply:
@zaro0508 i have this ticket out there:

https://sagebionetworks.jira.com/browse/IT-3931

My thought was to use an existing subdomain, however, i am also open to an entirely new domain for this as well. Im fine with whatever suggestion that you have for this.